updating geolocation api groupdata

chrisdavidmills opened the pull request:
As per https://bugzilla.mozilla.org/show_bug.cgi?id=1575144, geolocation has had the following interface naming updates:

 * Coordinates -> GeolocationCoordinates
 * Position -> GeolocationPosition
 * PositionError -> GeolocationPositionError

In addition, I added in the Geolocation API overview page, and linked to the "Using..." guide that I created.

chrisdavidmills commented:
@Elchi3 I know you've about the disappear on vacation; any chance you can fit in a quick review?

Elchi3 commented on macros/GroupData.json:
```diff
@@ -430,10 +430,12 @@
+                            "GeolocationCoordinates",
+                            "GeolocationPosition",
+                            "GeolocationPositionError",
                             "PositionOptions" ],
```
Honestly, I forgot what we've decided and implemented, but there is a "dictionaries" property now and `PositionOptions` is actually a dictionary, not an interface.

chrisdavidmills commented:
@Elchi3 cheers mate; I've moved the dictionary object inside the dictionaries member; I'm really glad you mentioned that, as it also highlighted some other fixes to make inside the docs themselves. Done those too!

And I changed the slug of the guide to "Using_the_Geolocation_API". It seemed a bit repetitive to me, hence using just "Using" initially, but I think the update is better for human readability, and probably also SEO too.
